content/posts/rsoc-2021-summary/index.md

```diff
@@ -0,0 +1,168 @@
+
+```
+$ rizin Project1.exe
+ -- Use scr.accel to browse the file faster!
+[0x00401703]> idpi ./Project1.pdb
+·········
+struct std::_Char_traits<char32_t,unsigned int> {
+        char32_t char_type;
+        uint32_t int_type;
+        int64_t off_type;
+        char32_t copy(char32_t * arg0, const char32_t * arg1, const uint32_t arg2);
+        char32_t _Copy_s(char32_t * arg0, const uint32_t arg1, const char32_t * arg2, const uint32_t arg3);
+        char32_t move(char32_t * arg0, const char32_t * arg1, const uint32_t arg2);
+        int32_t compare(const char32_t * arg0, const char32_t * arg1, uint32_t arg2);
+        uint32_t length(const char32_t * arg0);
+        const char32_t * find(const char32_t * arg0, uint32_t arg1, const char32_t * arg2);
+        bool eq(const char32_t * arg0, const char32_t * arg1);
+        bool lt(const char32_t * arg0, const char32_t * arg1);
+        char32_t to_char_type(const uint32_t * arg0);
+        uint32_t to_int_type(const char3_t * arg0);
+        bool eq_int_type(const uint32_t * arg0, const uint32_t * arg1);
+        uint32_t not_eof(const uint32_t * arg0);
+        uint32_t eof();
+ }
+·········
+[0x00401703]> tuc
+·········
+union __m64 {
+        uint64_t m64_u64;
+        float m64_f32[8];
+        unsigned char m64_i8[8];
+        int16_t m64_i16[8];
+        int32_t m64_i32[8];
+        int64_t m64_i64;
+        unsigned char m64_u8[8];
+        uint16_t m64_u16[8];
+        uint32_t m64_u32[8];
+};
+·········
+```
+Currently, Heersin has completed the new [RzIL](https://github.com/rizinorg/rizin/pull/1663), but it still lacks support for many architectures. So I am now [porting the 8051 architecture](https://github.com/rizinorg/rizin/pull/1609) from the old ESIL to the new RzIL, and I will be working with Heersin to port more architectures to the new IL afterwards.
+
+During this RSoC, I grew a lot and learned a lot of development skills that I would not normally be exposed to. I would like to especially thank my mentor Anton Kochkov for his selfless help. I would also like to thank all the community members for their help!
+
+# Heersin: New Rizin IL
+Hi, I'm Heersin, I particpated in RSoC this summer to introduce a new Intermediate Language and refactor ESIL related code. Rizin previously used ESIL([a stack-based IL](https://book.rizin.re/disassembling/esil.html)) as its IL to analyse binary. In fact, ESIL is neither user friendly nor developer friendly, that's the reason we don't like it. We take [BAP's Core Theory](http://binaryanalysisplatform.github.io/bap/api/odoc/bap-core-theory/Bap_core_theory/index.html) as our new IL. Because it's designed to be similar to SMT, and it may be the latest IL (the most "fashionable" one) we can trust for now.
```

Comment:
```suggestion
Hi, I'm Heersin, I particpated in RSoC this summer to introduce a new Intermediate Language and refactor ESIL related code. Rizin previously used ESIL([a stack-based IL](https://book.rizin.re/disassembling/esil.html)) as its IL to analyse binary. In fact, ESIL is neither user friendly nor developer friendly, those are some of the reasons that led to this work. We take [BAP's Core Theory](http://binaryanalysisplatform.github.io/bap/api/odoc/bap-core-theory/Bap_core_theory/index.html) as our new IL. Because it's designed to be similar to SMT, and it may be the latest IL (the most "fashionable" one) we can trust for now.
```